django/core/files/storage.py

```diff
@@ -248,6 +250,14 @@ def _value_or_setting(self, value, setting):
 
     @cached_property
     def base_location(self):
+        # Check if django test is running and if test media root is present.
+        if sys.argv[1] == "test":
```

Comment:
This is a very fragile way to find if we are running tests (consider someone writing their own test command by inheriting Django's command, and giving it a different name).
Please take inspiration from the way this is done for databases or mails, namely, instead of the specific parts of Django trying to include support for tests, the Test Runner sets things up in different parts of Django.